serviceToXML/src/main/java/tkhal/service/stringToXMLController/CustomerFactory.java

```diff
@@ -0,0 +1,111 @@
+package tkhal.service.stringToXMLController;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+
+public class CustomerFactory {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CustomerFactory.class);
+
+    public static LinkedList<Customer> create(String message) {
+        LinkedList<Customer> list = new LinkedList<>();
+        String[] records = message.split("\n");
+        for (String record : records) {
+            String symbol = record.substring(record.length()-1);
+            if (symbol.equals(",")) {
+                record += "?";
+            }
+            String[] value = record.split(",");
+            if (value[18].equals("\"")) {
+                value[17] += ",\"";
+                value = ArrayUtils.removeElement(value, value[18]);
+            }
+            if (value.length < 23) {
+                LOGGER.error("Invalid message \n" + Arrays.toString(value));
+            } else {
+                Customer customer = new Customer();
+                if (!(value[0].equals("?"))) customer.setAirline(value[0]);
+                if (!(value[1].equals("?"))) customer.setFlightNumber(value[1]);
+                if (!(value[2].equals("?"))) customer.setAcType(value[2]);
+                if (!(value[3].equals("?"))) customer.setValidTime(value[3]);
+                if (!(value[4].equals("?"))) customer.setCrossModelId(value[4]);
+                if (!(value[5].equals("?"))) customer.setTailNumber(value[5]);
+                if (!(value[6].equals("?"))) customer.setDepIcaoSiteId(value[6]);
+                if (!(value[7].equals("?"))) customer.setDepIcaoSiteId(value[7]);
+                if (!(value[8].equals("?") || value[8].length() == 0)) try {
+                    customer.setCopyNum(Integer.parseInt(value[8]));
+                } catch (NumberFormatException e) {
+                    LOGGER.error("error in element copyNum" + e);
+                }
+                if (!(value[9].equals("?") || value[9].length() == 0)) try {
```

Comment:
a lot of duplicated code. Pay attention on if statements

Reply:
made boolean function for general if statements